packages/cheerio-crawler/src/internals/cheerio-crawler.ts

```diff
@@ -1,40 +1,16 @@
-import { addTimeoutToPromise, tryCancel } from '@apify/timeout';
-import { concatStreamToBuffer, readStreamToString } from '@apify/utilities';
-import type { AutoscaledPoolOptions, BasicCrawlerOptions, ErrorHandler, RequestHandler } from '@crawlee/basic';
-import { BasicCrawler, BASIC_CRAWLER_TIMEOUT_BUFFER_SECS } from '@crawlee/basic';
-import type { CrawlingContext, EnqueueLinksOptions, ProxyConfiguration, Request, RequestQueue, Session } from '@crawlee/core';
-import { CrawlerExtension, enqueueLinks, mergeCookies, Router, resolveBaseUrlForEnqueueLinksFiltering, validators, Configuration } from '@crawlee/core';
+import type { BasicCrawlerOptions, ErrorHandler, RequestHandler } from '@crawlee/basic';
+import type { HttpCrawlerOptions } from '@crawlee/http';
+import { HttpCrawler } from '@crawlee/http';
+import type { CrawlingContext, EnqueueLinksOptions, ProxyConfiguration, RequestQueue, Configuration } from '@crawlee/core';
+import { enqueueLinks, Router, resolveBaseUrlForEnqueueLinksFiltering } from '@crawlee/core';
 import type { BatchAddRequestsResult, Awaitable, Dictionary } from '@crawlee/types';
 import type { CheerioRoot } from '@crawlee/utils';
-import { entries, parseContentTypeFromResponse } from '@crawlee/utils';
 import type { CheerioOptions } from 'cheerio';
 import * as cheerio from 'cheerio';
-import type { RequestLike, ResponseLike } from 'content-type';
-import contentTypeParser from 'content-type';
-import type { OptionsInit, Method, Request as GotRequest, Response as GotResponse, GotOptionsInit } from 'got-scraping';
-import { gotScraping, TimeoutError } from 'got-scraping';
+import type { OptionsInit, Response as GotResponse, GotOptionsInit } from 'got-scraping';
 import { DomHandler } from 'htmlparser2';
 import { WritableStream } from 'htmlparser2/lib/WritableStream';
-import type { IncomingHttpHeaders, IncomingMessage } from 'http';
-import iconv from 'iconv-lite';
-import ow from 'ow';
-import util from 'util';
-
-/**
- * Default mime types, which CheerioScraper supports.
- */
-const HTML_AND_XML_MIME_TYPES = ['text/html', 'text/xml', 'application/xhtml+xml', 'application/xml'];
-const APPLICATION_JSON_MIME_TYPE = 'application/json';
-const CHEERIO_OPTIMIZED_AUTOSCALED_POOL_OPTIONS: AutoscaledPoolOptions = {
```

Comment:
this needs to stay, we want different defaults for cheerio (and probably all the http crawlers)